### PR TITLE
DW-5237 added logic to trigger encryption script but not complete

### PR DESCRIPTION
**Pull request by sandeepg-github**

### CODE REVIEW

**Review by benjaminsherwood — DISMISSED** [review]

File: files/tarball_ingester_userdata.tpl [hunk 1]
```diff
@@ -80,15 +80,43 @@ aws s3 cp s3://${s3_artefact_bucket}/dataworks-tarball-ingester/dataworks-tarbal
...
-echo "Process data script would have run" >> /var/log/tarball_ingestion/tarball_ingestion.out 2>&1
+echo "Execute Python script to process Incrementals collections data..."
+python3 /opt/tarball_ingestion/steps/copy_collections_to_s3.py -s "${ti_src_dir}" \
+    -s3b "${ti_s3_bucket}" \
+    -s3p "${ti_s3_prefix}" \
+    -m "${ti_manifest_path}" \
+    -t "${ti_tmp_dir}" \
+    -d "${dks_endpoint}" \
+    -dt "${ti_dt}" \
+    -f "${ti_format}" \
```
Comment (sandeepg-github) [inline files/tarball_ingester_userdata.tpl]: should we hardcode the values for this to `Incrementals`?

File: files/tarball_ingester_userdata.tpl [hunk 2]
```diff
@@ -80,15 +80,43 @@ aws s3 cp s3://${s3_artefact_bucket}/dataworks-tarball-ingester/dataworks-tarbal
...
+
+echo "Execute Python script to process Full collections data..."
+python3 /opt/tarball_ingestion/steps/copy_collections_to_s3.py -s "${ti_src_dir}" \
+    -s3b "${ti_s3_bucket}" \
+    -s3p "${ti_s3_prefix}" \
+    -m "${ti_manifest_path}" \
+    -t "${ti_tmp_dir}" \
+    -d "${dks_endpoint}" \
+    -dt "${ti_dt}" \
+    -f "${ti_format}" \
```
Comment (sandeepg-github) [inline files/tarball_ingester_userdata.tpl]: should we hardcode the values for this to `Fulls`?

File: tarball-ingester.tf [hunk 3]
```diff
@@ -17,6 +17,17 @@ resource "aws_launch_template" "tarball_ingester" {
...
+    ti_src_dir                                       = local.ti_src_dir
+    ti_s3_bucket                                     = local.ti_s3_bucket
+    ti_s3_prefix                                     = local.ti_s3_prefix
+    ti_dks_url                                       = local.dks_endpoint
+    ti_format                                        = local.ti_format
+    ti_dt                                            = local.ti_dt
+    ti_wait                                          = local.ti_wait
+    ti_interval                                      = local.ti_interval
+    ti_manifest_path                                 = local.ti_manifest_path
+    ti_asg                                           = aws_autoscaling_group.tarball_ingester.id
```
Comment (sandeepg-github) [inline tarball-ingester.tf]: this should be `ti_asg_name` right? as per line 122 in user data file

File: files/tarball_ingester_userdata.tpl (hunk 1 above)
Comment (benjaminsherwood) [reply files/tarball_ingester_userdata.tpl]: Yes, in {{locals.tf}}

File: files/tarball_ingester_userdata.tpl (hunk 2 above)
Comment (benjaminsherwood) [reply files/tarball_ingester_userdata.tpl]: Yes, in {{locals.tf}}

File: tarball-ingester.tf (hunk 3 above)
Comment (benjaminsherwood) [reply tarball-ingester.tf]: Correct, will fix. Though it has another issue atm.

File: files/tarball_ingester_manifest.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    "mongoccalcdb-backup00.prod.ucds.io"
+  ],
+  "fulls": [
+    "mongoccalcdb-backup00.prod.ucds.io",
+    "mongocpdb-backup00.prod.ucds.io",
+    "mongocclaimdb-backup00.prod.ucds.io",
+    "mongocsdb-backup00.prod.ucds.io",
+    "mongoccdb-backup00.prod.ucds.io"
+  ]
+}
```
Comment (sandeepg-github) [inline files/tarball_ingester_manifest.json]: please add new line

File: files/tarball_ingester_userdata.tpl [hunk 5]
```diff
@@ -80,15 +81,47 @@ aws s3 cp s3://${s3_artefact_bucket}/dataworks-tarball-ingester/dataworks-tarbal
...
+yum install -y python3
+
 if [[ "${environment_name}" != "production" ]]; then
     echo "Running script to copy synthetic tarballs..."
     echo "Synthetic tarball script would have run" >> /var/log/tarball_ingestion/tarball_ingestion.out 2>&1
 fi
 
-echo "Execute Python script to process data..."
-echo "Process data script would have run" >> /var/log/tarball_ingestion/tarball_ingestion.out 2>&1
+TI_ASG_NAME=$(aws autoscaling describe-auto-scaling-instances \
```
Comment (sandeepg-github) [inline files/tarball_ingester_userdata.tpl:98]: nice i got what you did, this is outside tf realm

File: files/tarball_ingester_userdata.tpl [hunk 6]
```diff
@@ -80,15 +81,47 @@ aws s3 cp s3://${s3_artefact_bucket}/dataworks-tarball-ingester/dataworks-tarbal
...
+
+
+echo "Execute Python script to process Full collections data..."
+python3 /opt/tarball_ingestion/steps/copy_collections_to_s3.py -s "${ti_src_dir}" \
+    -s3b "${ti_s3_bucket}" \
+    -s3p "${ti_s3_prefix}" \
+    -m "${ti_manifest_path}" \
+    -t "${ti_tmp_dir}" \
+    -d "${dks_endpoint}" \
+    -f "full" \
```
Comment (sandeepg-github) [inline files/tarball_ingester_userdata.tpl]: this should be fulls,  as discussed this should represnt the key in manifest file

File: tarball-ingester-iam.tf [hunk 7]
```diff
@@ -166,6 +166,18 @@ data "aws_iam_policy_document" "minio_credentials_secretsmanager" {
...
 
+data "aws_iam_policy_document" "tarball_ingester_describe_autoscaling" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "autoscaling:DescribeAutoScalingInstances",
+    ]
+
+    resources = ["*"]
```
Comment (sandeepg-github) [inline tarball-ingester-iam.tf:177]: does it need to be "*" not sure if you can make it fine grained?

Comment (benjaminsherwood) [reply tarball-ingester-iam.tf:177]: tried it finegrained, it didn't like it, hence using *. as discussed with @dan-hill2802 it's a minimal risk.

**Review by benjaminsherwood — APPROVED** [review]

**Pull request merged**